diff --git a/components/lua/lmathlib.c b/components/lua/lmathlib.c
index 2696765..df1d8b3 100644
--- a/components/lua/lmathlib.c
+++ b/components/lua/lmathlib.c
@@ -174,8 +174,6 @@ static int math_pow (lua_State *L) {
 #endif
 
 
-#ifndef LUA_NUMBER_INTEGRAL
-#if 0
 static int math_log (lua_State *L) {
   lua_pushnumber(L, log(luaL_checknumber(L, 1)));
   return 1;
@@ -186,6 +184,10 @@ static int math_log10 (lua_State *L) {
   return 1;
 }
 
+#ifndef LUA_NUMBER_INTEGRAL
+#if 0
+
+
 static int math_exp (lua_State *L) {
   lua_pushnumber(L, exp(luaL_checknumber(L, 1)));
   return 1;
@@ -337,8 +339,8 @@ LROT_PUBLIC_BEGIN(math)
 // LROT_FUNCENTRY( mod, math_fmod )
 // LROT_FUNCENTRY( frexp, math_frexp )
 // LROT_FUNCENTRY( ldexp, math_ldexp )
-// LROT_FUNCENTRY( log10, math_log10 )
-// LROT_FUNCENTRY( log, math_log )
+  LROT_FUNCENTRY( log10, math_log10 )
+  LROT_FUNCENTRY( log, math_log )
   LROT_FUNCENTRY( max, math_max )
   LROT_FUNCENTRY( min, math_min )
 // LROT_FUNCENTRY( modf, math_modf )
